test/mpi/configure.ac

```diff
@@ -423,7 +394,7 @@ fi
 
 # errordir is substituted into the testlist file as errors when the
 # tests should check error handling and as a comment (#) otherwise.
-errordir="#"
+errordir="#errors"
```

Comment:
This is counter-intuitive to me. You can easily disable tests by removing them from a testlist file, but directories need to be defined and commented exactly like this to disable.

Reply:
> This is counter-intuitive to me. You can easily disable tests by removing them from a testlist file, but directories need to be defined and commented exactly like this to disable.

It is one way or the other. We either require explicit inclusion, or we require explicit exclusion. Since we intuitively want to have tests included, it makes sense to me to require explicit exclusion. And consider nested directories, the explicit inclusion requires multiple entries for a single sub-directory while the exclusion only need one at the right place.

The current patch is a compromise so the testlist files works for both scheme. Once transitioned, I expect all included directories in testlist files to disappear and you only see entries that you want to explicitly disable. The syntax can be relaxed and flexible. What syntax would you find intuitive?

By the way, the current configure using only '#' (or worse, empty) to remove directories is annoying. Numerous times I need check a particular test job on what is disabled, and each time I have to open both `testlist` and `testlist.in` to compare. I rarely need to check what is included since included tests are listed in the test report.

Reply:
> And consider nested directories, the explicit inclusion requires multiple entries for a single sub-directory while the exclusion only need one at the right place.

This is already possible now. Just remove or comment the directory in `testlist`.

> The current patch is a compromise so the testlist files works for both scheme. Once transitioned, I expect all included directories in testlist files to disappear and you only see entries that you want to explicitly disable. The syntax can be relaxed and flexible. What syntax would you find intuitive?

To me, it's not intuitive that directories and individual tests must be treated differently. When running by hand, it's simple to reduce test coverage by deleting or commenting entires in a `testlist` file until you are left with exactly what you want to run.

Reply:
> To me, it's not intuitive that directories and individual tests must be treated differently.

Directory and individual tests are different, right? Only the latter are actual testing subjects, the former are just container for managing tests. Anyway, I guess there is no point in arguing subjective intuitions. :)

The direct benefit for this PR is, if you want to run gpu tests only, all you need is to set `TESTLIST=testlist.gpu`, either with runtests command line or environment variables.

Reply:
I agree with the end goal of the PR. However the implementation is disruptive to a workflow I use often. Perhaps I am the only one who objects. Since @pavanbalaji was the one who requested the GPU-only test configuration, I would like to hear his opinion on this PR and #4645 in hopes we can find some way to move forward.

Reply:
I agree with Ken.  Treating directories and files differently in how they can be included/excluded is counter-intuitive and is a maintenance hassle.  All I was looking for is a simple way to speed up the GPU tests.  If this is becoming too problematic, another approach is to add a simple xfail for all other tests in the "gpu" jenkins configuration.  Let's not overcomplicate the test suite for this simple thing.

Reply:
> However the implementation is disruptive to a workflow I use often.

What is your workflow?

Reply:
> > However the implementation is disruptive to a workflow I use often.
> 
> What is your workflow?

I often modify `testlist` files by hand to run a subset of tests.

Reply:
> > > However the implementation is disruptive to a workflow I use often.
> > 
> > 
> > What is your workflow?
> 
> I often modify `testlist` files by hand to run a subset of tests.

You still can, right? Simply place a '#' in front of the line (including directories) that you want to exclude?

Reply:
> You still can, right? Simply place a '#' in front of the line (including directories) that you want to exclude?

Yes, of course. But as I said, I find this unintuitive. It requires the user to treat directories differently than before, when you could simply delete everything you didn't want to run.

> The current PR is a compromise for both practice. You can either do the old way explicitly commenting out directory; or there is a new way, simply create a testlist.custom and run that directly.

But you can't delete directories. The thing I did most often will no longer work.

Reply:
I see. You prefer "delete", rather than "comment out".